code/hsec-tools/src/Security/OSV.hs

```diff
@@ -0,0 +1,223 @@
+-- | This module contains the OSV datatype and its ToJSON instance.
+-- The module was initialized with http://json-to-haskell.chrispenner.ca/
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Security.OSV where
+
+import Data.Aeson (ToJSON(..), FromJSON(..), Value(..), (.:), (.=), object)
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Text (Text)
+
+data Affected = Affected
+  { affectedRanges :: [Ranges]
+  , affectedPackage :: Package
+  , affectedEcosystemSpecific :: EcosystemSpecific
+  , affectedDatabaseSpecific :: DatabaseSpecific
+  } deriving (Show, Eq, Ord)
+
+data Affects = Affects
+  { affectsOs :: [Value]
+  , affectsArch :: [Value]
+  , affectsFunctions :: [Value]
+  } deriving (Show, Eq, Ord)
+
+data DatabaseSpecific = DatabaseSpecific
+  { databaseSpecificCvss :: Maybe Value
+  , databaseSpecificCategories :: [Value]
+  , databaseSpecificInformational :: Text
+  } deriving (Show, Eq, Ord)
+
+data EcosystemSpecific = EcosystemSpecific
+  { ecosystemSpecificAffects :: Affects
+  } deriving (Show, Eq, Ord)
+
+data Events = Events
+  { eventsIntroduced :: Text
+  } deriving (Show, Eq, Ord)
+
+data Model = Model
+  { modelDetails :: Text
+  , modelId :: Text
+  , modelSummary :: Text
+  , modelRelated :: [Value]
+  , modelAffected :: [Affected]
+  , modelAliases :: [Value]
+  , modelPublished :: Text
+  , modelReferences :: [References]
+  , modelSeverity :: [Value]
+  , modelModified :: Text
+  } deriving (Show, Eq, Ord)
+
+data Package = Package
+  { packageName :: Text
+  , packageEcosystem :: Text
+  , packagePurl :: Text
+  } deriving (Show, Eq, Ord)
+
+data Ranges = Ranges
+  { rangesEvents :: [Events]
+  , rangesType :: Text
```

Comment:
I think we're missing `repo` here

https://ossf.github.io/osv-schema/#affectedrangesrepo-field